src/System.Runtime/tests/System/Enum.cs

```diff
@@ -4,219 +4,165 @@
-
-            s = Enum.GetName(t, 1);
-            Assert.Equal(s, "Red");
-
-            s = Enum.GetName(t, SimpleEnum.Red);
-            Assert.Equal(s, "Red");
-
-            // In the case of multiple matches, GetName returns one of them (which one is an implementation detail.)
-            s = Enum.GetName(t, 3);
-            Assert.True(s == "Green" || s == "Green_a" || s == "Green_b");
+            b = Enum.TryParse<SimpleEnum>(value, out e);
+            Assert.Equal(expectedCanParse, b);
+            if (expectedCanParse)
+            {
+                Assert.Equal(expectedParseResult, e);
+            }
         }
 
-        // Negative tests
+        b = Enum.TryParse<SimpleEnum>(value, ignoreCase, out e);
+        Assert.Equal(expectedCanParse, b);
+        if (expectedCanParse)
         {
-            Type t = typeof(SimpleEnum);
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(null, 1));
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(t, null));
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(typeof(Object), null));
-            Assert.Throws<ArgumentException>(() => Enum.GetName(t, "Red"));
-            Assert.Throws<ArgumentException>(() => Enum.GetName(t, (IntPtr)0));
+            Assert.Equal(expectedParseResult, e);
         }
+    }
 
-        {
-            /*
-             * Despite what MSDN says, GetName() does not require passing in the exact integral type.
-             * 
-             * For the purposes of comparison:
-             * 
-             *  - The enum member value are normalized as follows:
-             *    - unsigned ints zero-extended to 64-bits
-             *    - signed ints sign-extended to 64-bits
-             *
-             *  - The value passed in as an argument to GetNames() is normalized as follows:
-             *    - unsigned ints zero-extended to 64-bits
-             *    - signed ints sign-extended to 64-bits
-             *
-             *  Then comparison is done on all 64 bits.
-             */
-
-            Type t = typeof(SByteEnum);
-            s = Enum.GetName(t, 0xffffffffffffff80LU);
-            Assert.Equal(s, "Min");
-
-            s = Enum.GetName(t, 0xffffff80u);
-            Assert.Equal(s, null);
-
-            s = Enum.GetName(t, unchecked((int)(0xffffff80u)));
-            Assert.Equal(s, "Min");
-
-            s = Enum.GetName(t, true);
-            Assert.Equal(s, "One");
-
-            s = Enum.GetName(t, (char)1);
-            Assert.Equal(s, "One");
-
-            // The api doesn't even care if you pass in a completely different enum!
-            s = Enum.GetName(t, SimpleEnum.Red);
-            Assert.Equal(s, "One");
-        }
-        return;
+    [Theory]
+    [InlineData(99, null)]
+    [InlineData(1, "Red")]
+    [InlineData(SimpleEnum.Red, "Red")]
+    public static void TestGetName(object value, string expected)
+    {
+        String s = Enum.GetName(typeof(SimpleEnum), value);
```

Comment:
Nit: String => string.  I realize this file is inconsistent, but as long as you're changing the code that uses it, you can use the "right" one, especially since it's lowercase in the method signature.